File: src/0347-前K个高频元素/0347.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        time_dict = {}
        for num in nums:
            time_dict[num] = time_dict.get(num,0) + 1
        
        res_list = sorted(time_dict.items(), key = lambda x:x[1], reverse = True)
        
        ans  = []
        for i in range(0,k):
            ans.append(res_list[i][0])
        return ans

File: src/0347-前K个高频元素/0347.cpp
class Solution
{
public:
    vector<int> topKFrequent(vector<int> &nums, int k)
    {
        // 找最大count
        unordered_map<int, int> counts;
        int max_count = 0;
        for (const int &num : nums)
        {
            max_count = max(max_count, ++counts[num]);
        }

        // 把每个数组元素的数量放入buckets
        vector<vector<int>> buckets(max_count + 1);
        for (const auto &p : counts)
        {
            buckets[p.second].push_back(p.first);
        }

        vector<int> ans;
        for (int i = max_count; i >= 0 && ans.size() < k; i--)
        {
            for (const int &num : buckets[i])
            {
                ans.push_back(num);
                if (ans.size() == k)
                {
                    break;
                }
            }
        }

        return ans;
    }
};

//与其说是桶排序，不如说是反向索引